compute-timing-stats

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/env ruby
+# A script to analyze a range of CircleCI builds for timing statistics
+# For available options, run ./compute-timing-stats -h
+require 'fileutils'
+require 'launchy'
+require 'optparse'
+require 'time'
+require 'pp'
+require_relative './lib/circle_project'
+
+OUTPUT_DIRECTORY = File.expand_path('~/.circle-cli/output').freeze
+
+def main(options)
+
+  # Download and format the information we need
+  @project = CircleProject.new('code-dot-org/code-dot-org')
+  build_range = @project.build_range_from_options(options)
+
+  if options[:output_format] == :plot
+    builds = []
+    step_times = {}
+    @project.get_builds(build_range, true).each do |build|
+      next if build['outcome'] != 'success'
+      next if build['steps'].nil? || build['build_time_millis'].nil?
+      next unless options[:branch_name].nil? || build['branch'] == options[:branch_name]
+
+      build_steps = {}
+      build['steps'].each do |step|
+        next unless options[:step_pattern].nil? || options[:step_pattern] =~ step['name']
+        step['actions'].each do |action|
+          next unless options[:container_num] == action['index'] || step['actions'].size == 1
+          step_name = step['name']
+          step_duration = action['run_time_millis'] / 1000 # in seconds
```

Comment:
nit: Maybe we have guidelines already, but feels like more spaces before the # might be a bit more readable.